rasa/core/policies/embedding_policy.py

```diff
@@ -638,57 +353,204 @@ def load(cls, path: Text) -> "EmbeddingPolicy":
+            self.config[C2],
+            "label",
+            self.config[SIMILARITY_TYPE],
+        )
 
-            a_in = train_utils.load_tensor("user_placeholder")
-            b_in = train_utils.load_tensor("bot_placeholder")
+    def _create_all_labels_embed(self) -> Tuple[tf.Tensor, tf.Tensor]:
+        all_label = tf.constant(
+            self._encoded_all_label_ids, dtype=tf.float32, name="all_label"
+        )
+        all_labels_embed = self._embed_label(all_label)
 
-            sim_all = train_utils.load_tensor("similarity_all")
-            pred_confidence = train_utils.load_tensor("pred_confidence")
-            sim = train_utils.load_tensor("similarity")
+        return all_label, all_labels_embed
 
-            dial_embed = train_utils.load_tensor("dial_embed")
-            bot_embed = train_utils.load_tensor("bot_embed")
-            all_bot_embed = train_utils.load_tensor("all_bot_embed")
+    def _emebed_dialogue(self, dialogue_in: tf.Tensor) -> Tuple[tf.Tensor, tf.Tensor]:
+        """Create dialogue level embedding and mask."""
 
-            attention_weights = train_utils.load_tensor("attention_weights")
+        # mask different length sequences
+        # if there is at least one `-1` it should be masked
+        mask = tf.sign(tf.reduce_max(dialogue_in, -1) + 1)
 
-        return cls(
-            featurizer=featurizer,
-            priority=meta.pop("priority"),
-            graph=graph,
-            session=session,
-            user_placeholder=a_in,
-            bot_placeholder=b_in,
-            similarity_all=sim_all,
-            pred_confidence=pred_confidence,
-            similarity=sim,
-            dial_embed=dial_embed,
-            bot_embed=bot_embed,
-            all_bot_embed=all_bot_embed,
-            attention_weights=attention_weights,
-            **meta,
+        dialogue = self._tf_layers["ffnn.dialogue"](dialogue_in, self._training)
+        dialogue_transformed = self._tf_layers["transformer"](
+            dialogue, tf.expand_dims(mask, axis=-1), self._training
         )
+
+        if self.max_history_tracker_featurizer_used:
+            # pick last label if max history featurizer is used
+            dialogue_transformed = dialogue_transformed[:, -1:, :]
+            mask = mask[:, -1:]
+
+        dialogue_embed = self._tf_layers["embed.dialogue"](dialogue_transformed)
+
+        return dialogue_embed, mask
+
+    def _embed_label(self, label_in: tf.Tensor) -> tf.Tensor:
+        label = self._tf_layers["ffnn.label"](label_in, self._training)
+        return self._tf_layers["embed.label"](label)
+
+    def batch_loss(self, batch_in: List[tf.Tensor]) -> tf.Tensor:
+        dialogue_in, label_in, _ = batch_in
+
+        if self.max_history_tracker_featurizer_used:
+            # add time dimension if max history featurizer is used
+            label_in = label_in[:, tf.newaxis, :]
+
+        all_label, self.all_labels_embed = self._create_all_labels_embed()
```

Comment:
there is no need for `self.` in training, it is recalculated every time anyhow
```suggestion
        all_label, all_labels_embed = self._create_all_labels_embed()
```